server/aesdsocket.c

```diff
@@ -0,0 +1,328 @@
+               // printf("Calloc:\n");
+                perror("Calloc fail :");
+                return -1;
+            }
+            //printf("calloc done\n");
+            memset(store_buffer, '\0', INIT_BUFFER_SIZE);
+
+            total_length = 0;
+            
+            struct sockaddr_in *temp_client_addr = (struct sockaddr_in*) &client_addr;    
+            // log connected to client 
+           // printf("Connected to client %s \n", inet_ntoa(temp_client_addr->sin_addr));
+            syslog(LOG_DEBUG, "Connected to client %s", inet_ntoa(temp_client_addr->sin_addr));
+            
+            // loop until client closes connection or \n is found
+            while(((client_rx = recv(client_fd, rx_buffer, sizeof(rx_buffer), 0)) > 0) && (!(eol_found))) {
+                
+               // printf("received data %s\n", rx_buffer);
+                line_length = 0;
+
+                for(int i = 0; i < client_rx; i++) {
+                    line_length++;
+                    if(rx_buffer[i] == '\n') {
+                        eol_found = true;
+                     //   printf("eof found at %d\n", line_length);
+                        break;
+                    }
+                    
+                }
+
+                total_length += line_length;
+
+                if(total_length >= INIT_BUFFER_SIZE) { // if total store buffer is falling short, realloc
+
+                    if(!(new_ptr = realloc(store_buffer, INIT_BUFFER_SIZE + total_length + 1))) {
+                     //   printf("realloc :\n");    
+                        perror("Realloc fail :");
+                        return -1;
+                    } else { // append to the string
+                        store_buffer = new_ptr;
+                        strncat(store_buffer, rx_buffer, line_length);
+                      //  printf("%s concat\n", store_buffer);
+                    }
+                
+                } else { // eol found, leave recv while and append to string 
+                    strncat(store_buffer, rx_buffer, line_length);
+                }
+
+                if(eol_found) {
+                    break;
+                }
+                
+            }
+            
+            
+            // seek to the EOF
+            off_t total_file_length = 0;
+            if((total_file_length = lseek(fd, 0, SEEK_END))== -1) {
+                perror("lseek error: ");
+                return -1;
+
+            }
+            if(write(fd, store_buffer, total_length) == -1) {
+                perror("File writer fail: ");
+                return -1;
+            }
+            
+            free(store_buffer);
+
+           // long total_file_length = ftell(fd); 
+            if((total_file_length = lseek(fd, 0, SEEK_END))== -1) {
+                perror("lseek error: ");
+                return -1;
+
+            }
+            
+            char *send_buffer = (char *) malloc(sizeof(char) * total_file_length);
```

Comment:
@sachininja what assumption are you making here about the total size of the heap and the total size of all packets sent?

Reply:
I have fixed this with A6 release with a fixed buffer size